src/coreclr/interpreter/compiler.cpp

```diff
@@ -2854,6 +2854,21 @@ void InterpCompiler::EmitPushLdvirtftn(int thisVar, CORINFO_RESOLVED_TOKEN* pRes
     m_pLastNewIns->info.pCallInfo->pCallArgs = callArgs;
 }
 
+void InterpCompiler::EmitCalli(bool isTailCall, void* calliCookie, int callIFunctionPointerVar, CORINFO_SIG_INFO* callSiteSig)
+{
+    AddIns(isTailCall ? INTOP_CALLI_TAIL : INTOP_CALLI);
+    m_pLastNewIns->data[0] = GetDataItemIndex(calliCookie);
+    // data[1] is set to 1 if the calli is calling a pinvoke, 0 otherwise
+    bool suppressGCTransition = false;
+    m_compHnd->getUnmanagedCallConv(nullptr, callSiteSig, &suppressGCTransition);
+    bool isPInvoke = (callSiteSig->callConv != CORINFO_CALLCONV_DEFAULT && callSiteSig->callConv != CORINFO_CALLCONV_VARARG);
+    bool isPInvokeMarshalled = isPInvoke && m_compHnd->pInvokeMarshalingRequired(NULL, callSiteSig);
+    m_pLastNewIns->data[1] = (suppressGCTransition ? (int32_t)CalliFlags::SuppressGCTransition : 0) |
+                             (isPInvoke ? (int32_t)CalliFlags::PInvoke : 0) |
+                             (isPInvokeMarshalled ? (int32_t)CalliFlags::PInvokeMarshalled : 0);
```

Comment:
The handling for unmanaged calli with marshalling looks incomplete. The JIT needs to call `GetCookieForPInvokeCalliSig` for that - I do not see it called by the interpreter JIT.

Here is an example:
```csharp
unsafe void UnmanagedMarshalledCalli(void* p, string s)
{
     ((delegate* unmanaged<string, void>)p)(s);	 // The string should be marshalled as pinned pointer
}
```

I think it would be ok to not support calli with marshalling for now or for the interpreter in general. The only reason why we are supporting it is compatibility with managed C++. I doubt that it is used on mobile / browser. If you take this path, we should assert or throw exception to make it clear that it does not work.